Automattic/wp-calypso
# Decouple buttons

#### Changes proposed in this Pull Request

* The PR will try to decouple the login and subscribe buttons used in the non-subscriber view in the premium content block

Ref: https://github.com/Automattic/wp-calypso/issues/41850

I plan to use recent changes to [block context](https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/block-api/block-context.md) to allow nested blocks to know what the container's selected plan ID is so it can be used to determine if those blocks should be rendered or not.

#### Testing instructions

* Apply PR and make sure sync'd with sandbox.
* Go to test FSE site (that is sandboxed).
* Test adding a premium content block to a post.
* Test that the buttons are rendered ok in the editor and in the published post.
* Test that you can remove all buttons and add them again inside the logged out (non-subscriber) view - you shouldn't be able to add a Premium content button to the subscriber view OR outside the Premium content container.
